packages/regl-worldview/src/commands/GLTFScene.js

```diff
@@ -0,0 +1,210 @@
+    return undefined;
+  }
+  // prettier-ignore
+  switch (value) {
+    // min/mag filters
+    case WebGLRenderingContext.NEAREST: return 'nearest';
+    case WebGLRenderingContext.LINEAR: return 'linear';
+    case WebGLRenderingContext.NEAREST_MIPMAP_NEAREST: return 'nearest mipmap nearest';
+    case WebGLRenderingContext.NEAREST_MIPMAP_LINEAR: return 'nearest mipmap linear';
+    case WebGLRenderingContext.LINEAR_MIPMAP_NEAREST: return 'linear mipmap nearest';
+    case WebGLRenderingContext.LINEAR_MIPMAP_LINEAR: return 'linear mipmap linear';
+    // texture wrapping modes
+    case WebGLRenderingContext.REPEAT: return 'repeat';
+    case WebGLRenderingContext.CLAMP_TO_EDGE: return 'clamp';
+    case WebGLRenderingContext.MIRRORED_REPEAT: return 'mirror';
+  }
+  throw new Error(`unhandled constant value ${JSON.stringify(value)}`);
+}
+
+const drawModel = (regl) => {
+  const command = regl({
+    primitive: "triangles",
+    uniforms: {
+      baseColorTexture: regl.prop("baseColorTexture"),
+      nodeMatrix: regl.prop("nodeMatrix"),
+      poseMatrix: regl.context("poseMatrix"),
+      "light.direction": [0, 0, -1],
+      "light.ambientIntensity": 0.5,
+      "light.diffuseIntensity": 0.5,
+    },
+    attributes: {
+      position: regl.prop("positions"),
+      normal: regl.prop("normals"),
+      texCoord: regl.prop("texCoords"),
+    },
+    elements: regl.prop("indices"),
+    vert: `
+  uniform mat4 projection, view;
+  uniform mat4 nodeMatrix;
+  uniform mat4 poseMatrix;
+  attribute vec3 position, normal;
+  varying vec3 vNormal;
+  attribute vec2 texCoord;
+  varying vec2 vTexCoord;
+
+  void main() {
+    // using the projection matrix for normals breaks lighting for orthographic mode
+    mat4 mv = view * poseMatrix * nodeMatrix;
+    vNormal = normalize((mv * vec4(normal, 0)).xyz);
+    vTexCoord = texCoord;
+    gl_Position = projection * mv * vec4(position, 1);
+  }
+  `,
+    frag: `
+  precision mediump float;
+  uniform sampler2D baseColorTexture;
+  varying mediump vec2 vTexCoord;
+  varying mediump vec3 vNormal;
+
+  // Basic directional lighting from:
+  // http://ogldev.atspace.co.uk/www/tutorial18/tutorial18.html
+  struct DirectionalLight {
+    mediump vec3 direction;
+    lowp float ambientIntensity;
+    lowp float diffuseIntensity;
+  };
+  uniform DirectionalLight light;
+
+  void main() {
+    vec3 baseColor = texture2D(baseColorTexture, vTexCoord).rgb;
+    float diffuse = light.diffuseIntensity * max(0.0, dot(vNormal, -light.direction));
+    gl_FragColor = vec4((light.ambientIntensity + diffuse) * baseColor, 1);
+  }
+  `,
+  });
+
+  let drawCalls;
+  function prepareDrawCallsIfNeeded(model) {
+    if (drawCalls) {
+      return;
+    }
+
+    const textures = model.json.textures.map((texture) => {
+      const sampler = model.json.samplers[texture.sampler];
+      return regl.texture({
+        data: model.images[texture.source],
+        min: "linear", //glConstantToRegl(sampler.minFilter),
```

Comment:
what's the commented out part here for?

Reply:
For our car model this made some of the textures look bad when downsampling since they chose the “best” mip level. I don’t totally understand how mipmaps are supposed to work, so this is just a workaround that makes it look slightly better, although the downside is now it looks *too* sharp sometimes 🤷‍♂️

I guess I could add a prop to override this. Or maybe a prop to allow mutating/changing the model after it’s loaded...what do you think?